code/game/atoms.dm

```diff
@@ -20,6 +20,7 @@ var/global/list/ghdel_profiling = list()
 	var/had_blood //Something was bloody at some point.
 	var/germ_level = 0 // The higher the germ level, the more germ on the atom.
 	var/penetration_dampening = 5 //drains some of a projectile's penetration power whenever it goes through the atom
+	var/obj/item/device/grey_market_beacon/market_beacon = null //For the grey market.
```

Comment:
As far as I can tell, the only thing this var is used for is to tell if something already has a market beacon in it
Which is accomplished with just `if(locate(/obj/item/device/grey_market_beacon) in contents)`

Reply:
Saves on locate operations though

Reply:
Yes, it saves a locate operation when you examine something, at the cost of adding a var to all atoms

Reply:
This does not work since contents is used by many things for storage. For instance, lockers. While those are disabled now due to balancing, putting a beacon on a locker would allow you to open it and retrieve it.
And a single null memory address will not affect memory usage almost at all. I could go get rid of many of the other horrid atom-wide variables if you'd like.

Reply:
You're right; I wasn't paying enough attention. I don't like it but there isn't really a better way to do it.
Well, actually, unless you can attach the beacon to turfs it could be moved down to `/atom/movable`, but nothing other than that.

Reply:
I'll do that since I removed being able to teleport doors away.

Reply:
Doors are still `/atom/movable`, for the record
Everything except turfs and areas falls under that